compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -295,6 +295,22 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             sym::write_bytes => {
                 self.write_bytes_intrinsic(&args[0], &args[1], &args[2])?;
             }
+            sym::swap_nonoverlapping_single => {
+                self.swap_memory(
+                    self.read_pointer(&args[0])?,
+                    self.read_pointer(&args[1])?,
+                    1,
+                    self.deref_operand(&args[0])?.layout,
```

Comment:
Using `args[0]` twice here isn't good, each argument should be converted to its high-level representation exactly once.

Does `let layout = self.layout_of(substs.type_at(0))?` like in some of the other intrinsics help?